### PR TITLE
Static buffers

### PR DESCRIPTION
**Pull request by nirbhayc**
.

### CODE REVIEW

File: sqlite/comdb2build.c [hunk 1]
```diff
@@ -846,200 +860,260 @@ void comdb2RebuildIndex(Parse* pParse, Token* nm, Token* lnm, Token* index, int
...
+    Vdbe *v = sqlite3GetVdbe(pParse);
+
     if (comdb2AuthenticateUserOp(v, pParse))
-        return;     
-    TokenStr(name, nm);
-    if (strlen(name) >= MAX_SPNAME) {
-        sqlite3ErrorMsg(pParse, "bad procedure name:%s", name);
+        return;
+
+    if (comdb2TokenToStr(nm, spname, sizeof(spname))) {
```
Comment (mponomar) [inline sqlite/comdb2build.c]: Should this be dequoted?

File: sqlite/comdb2build.c [hunk 2]
```diff
@@ -1472,36 +1546,50 @@ void comdb2enableAuth(Parse* pParse, int on)
...
+    bpfunc_password__init(password);
 
     arg->pwd = password;
     arg->type = BPFUNC_PASSWORD;
     password->disable = 0;
-  
-    if (create_string_from_token(v, pParse, &password->user, nm) ||
-        create_string_from_token(v, pParse, &password->password, pwd))
-            goto clean_arg;
+
```
Comment (mponomar) [inline sqlite/comdb2build.c]: dequote?

File: sqlite/comdb2build.c [hunk 3]
```diff
@@ -1678,30 +1766,33 @@ void comdb2getAnalyzeThreshold(Parse* pParse, Token *nm, Token *lnm)
...
+void resolveTableName(struct SrcList_item *p, const char *zDB, char *tableName,
+                      size_t len)
 {
    struct sql_thread *thd = pthread_getspecific(query_info_key);
    if ((zDB && (!strcasecmp(zDB, "main") || !strcasecmp(zDB, "temp"))))
    {
-       sprintf(tableName, "%s", p->zName);
-   } else if (thd->clnt && (thd->clnt->user[0] != '\0') && !strstr(p->zName, "@")
-          && strncasecmp(p->zName, "sqlite_", 7) && strncasecmp(p->zName, "comdb2", 6))
+       snprintf(tableName, len, "%s", p->zName);
```
Comment (mponomar) [inline sqlite/comdb2build.c]: Check for truncation?

File: sqlite/comdb2build.c (hunk 1 above)
Comment (nirbhayc) [reply sqlite/comdb2build.c]: comdb2TokenToStr() also dequotes the token.

File: sqlite/comdb2build.c (hunk 2 above)
Comment (nirbhayc) [reply sqlite/comdb2build.c]: ditto

File: sqlite/comdb2build.c (hunk 3 above)
Comment (nirbhayc) [reply sqlite/comdb2build.c]: Done

File: sqlite/comdb2build.c [hunk 4]
```diff
@@ -846,200 +873,260 @@ void comdb2RebuildIndex(Parse* pParse, Token* nm, Token* lnm, Token* index, int
...
-    strncpy(tp->partition_name, partition_name->z, max_length);
+    if (!tp->partition_name) {
+        setError(pParse, SQLITE_NOMEM, "Out of Memory");
+        goto clean_arg;
+    }
+    if (partition_name->n >= MAXTABLELEN) {
+        setError(pParse, SQLITE_MISUSE, "Partition name is too long");
+        goto clean_arg;
+    }
+    strncpy0(tp->partition_name, partition_name->z, partition_name->n + 1);
```
Comment (ldennis) [inline sqlite/comdb2build.c]: maybe out of the scope of this PR, but I am wondering if we need dequote for partition name

Comment (ldennis) [reply sqlite/comdb2build.c]: time partition and friends seem to dequote only table name, i think it is good to leave it as it is for now.

**Comment by ldennis** [comment]
Looks good otherwise

File: sqlite/comdb2build.c [hunk 5]
```diff
@@ -1472,36 +1559,50 @@ void comdb2enableAuth(Parse* pParse, int on)
 
 void comdb2setPassword(Parse* pParse, Token* pwd, Token* nm)
 {
+    char username[MAX_USERNAME_LEN];
+    char passwd[MAX_PASSWORD_LEN];
+
     Vdbe *v  = sqlite3GetVdbe(pParse);
+
+    if (comdb2TokenToStr(pwd, passwd, sizeof(passwd))) {
```
[the remark this answers is not in the document]
Comment (nirbhayc) [reply sqlite/comdb2build.c]: The token needs to be dequoted before validating its length. comdb2TokenToStr does the validation  besides copying the dequoted token into the specified buffer.

**Pull request merged**